vagrant/Vagrantfile

```diff
@@ -970,7 +974,11 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
 			if vm.vm.box.start_with? 'atomic-' or vm_sync == ''
 				disabled = true
 			end
-			vm.vm.synced_folder "#{dot}", '/vagrant', disabled: disabled, type: vm_sync	# nfs, rsync
+			if vm_sync == 'nfs4'
+				vm.vm.synced_folder "#{dot}", '/vagrant', disabled: disabled, type: 'nfs', nfs_version: 4, nfs_udp: false
+			else
+				vm.vm.synced_folder "#{dot}", '/vagrant', disabled: disabled, type: vm_sync	# nfs, rsync
+			end
```

Comment:
For the above code, I wanted to dynamically construct the arguments type, nfs_version, and nfs_udp so I could not have to copy the other parameters. Sadly, my Ruby skills are nonexistent, and I'm coding on an airplane without Internet access for Ruby documentation. In Python, I'd do something like this:

```
params = {type: 'nfs'}
if vm_sync == "nfs4":
    params['nfs_version'] = 4
    params['nfs_udp'] = False
    function(some_stuff, **params)
```

If you can tell me how to do that I'll change it.